### PR TITLE
Refactor the contact manager in DEM

### PR DESCRIPTION
**Pull request by acdaigneault**
# Description of the problem

- The container manager class was more a contact manager with extra random containers and it was a bit weird.

# Description of the solution

- Replace the DEMContainerManager class to DEMContactManager class
- Move some containers back to DEM or CFD-DEM coupling
- Fix a small error in the map of floating walls since it used particle_id type for floating walls

# How Has This Been Tested?

- Tests are good

### CODE REVIEW

File: include/dem/dem.h [hunk 1]
```diff
@@ -360,8 +359,23 @@ class DEMSolver
...
+  typename dem_data_structures<dim>::floating_mesh_information
+    floating_mesh_info;
+  typename dem_data_structures<dim>::boundary_points_and_normal_vectors
+    updated_boundary_points_and_normal_vectors;
+  typename dem_data_structures<dim>::vector_on_boundary
+    forces_boundary_information;
+  typename dem_data_structures<dim>::vector_on_boundary
+    torques_boundary_information;
+  typename DEM::dem_data_structures<dim>::periodic_boundaries_cells_info
+    periodic_boundaries_cells_information;
```
Comment (blaisb) [inline include/dem/dem.h:373]: Should these types be defined here or could they be defined in another .h file to be more neutral?

File: source/dem/CMakeLists.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@ add_library(lethe-dem
   # Sources
   data_containers.cc
   dem.cc
-  dem_container_manager.cc
+        dem_contact_manager.cc
```
Comment (blaisb) [inline source/dem/CMakeLists.txt:5]: indent here is weird no?

**Review by blaisb — APPROVED** [review]
Some tiny comments, that's all. Great work!

File: include/dem/dem.h (hunk 1 above)
Comment (acdaigneault) [reply include/dem/dem.h:373]: Mhm I don't know, they are not really related to each other, it's again a bit random to me. What do you think?

Comment (blaisb) [reply include/dem/dem.h:373]: Good for me then :)!

**Pull request merged**